### PR TITLE
feat: revoke functional tests (AM-384)

### PR DESCRIPTION
**Pull request by bmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-384

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/amapi/functional/api/RevokeResourceAccessApiTest.java [hunk 1]
```diff
@@ -48,8 +62,108 @@ public void verifyRevokeExplicitAccessApiWithoutResourceNameAndServiceName() {
...
+            .resourceName("some resource")
+            .resourceType(resourceType)
+            .build();
+
+        Response response = createThenRevokeAccess(explicitAccessGrant, explicitAccessMetadata);
+
+        response.then()
+            .assertThat()
+            .statusCode(HttpStatus.NO_CONTENT.value())
+            .log();
```
[the remark this answers is not in the document]
Comment (shashank-rastogi) [reply src/functionalTest/java/uk/gov/hmcts/reform/amapi/functional/api/RevokeResourceAccessApiTest.java]: Is this impl correct? if you are revoking explicit Access with incorrect Resource Name and Service Name then should not it be setting wrong service name and resource name in the revoke API and checking in DB that no records are deleted?

[the remark this answers is not in the document]
Comment (bmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/amapi/functional/api/RevokeResourceAccessApiTest.java]: That is what happens here. A call to the revoke access endpoint with the incorrect resource name/service name should not delete anything from the access management table. This means that a subsequent call to the filter resource endpoint should return data based on that explicit access record as it still exists

[the remark this answers is not in the document]
Comment (bmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/amapi/functional/api/RevokeResourceAccessApiTest.java]: There is no way for the functional tests to access the DB to check that the explicit access record still exists, so using the filter resource method is a way of checking the existence of the explicit access record without directly connecting to the DB

[the remark this answers is not in the document]
Comment (shashank-rastogi) [reply src/functionalTest/java/uk/gov/hmcts/reform/amapi/functional/api/RevokeResourceAccessApiTest.java]: 👍 , i missed the code snippet below to check the record using FilterResource API

Good to go! you can ship it

**Pull request merged**